src/Agent/NewRelic/Agent/Core/OpenTelemetryBridge/MeterListenerBridge.cs

```diff
@@ -0,0 +1,1449 @@
+            }
+
+            // The MeasurementsCompleted event is triggered when an instrument is no longer being subscribed to by a MeterListener
+            // or the Meter holding the instrumented is Disposed. In our use case, we subscribe to all instruments in a meter and
+            // do not disable individual individual instruments, so the only way this event will be triggered is if the corresponding
+            // meter is no longer being used by the application or if our MeterListener is being Disposed. In either case we can
+            // just Dispose of the bridged Meter.
+
+            if (_bridgedMeters.TryRemove(bridgedInstrument.Meter.Name, out var bridgedMeter))
+            {
+                var meterName = bridgedMeter.Name;
+                bridgedMeter.Dispose();
+                Log.Debug($"Disposed bridged meter: {meterName}");
+            }
+        }
+
+
+        private static bool ShouldEnableInstrumentsInMeter(string meterName)
+        {
+            // TODO: If the agent ever targets .net 9 the .net 9 build of the agent will not need to bridge the runtime metrics meter that is
+            // built into the diagnostics source library. This check will need to be updated to not bridge the runtime metrics meter.
+
+            // Skip internal diagnostic meters to avoid potential recursion
+            if (string.IsNullOrEmpty(meterName))
+            {
+                return false;
+            }
+
+            // Filter out internal infrastructure meters that could cause recursion
+            if (meterName.StartsWith("NewRelic", StringComparison.OrdinalIgnoreCase) ||
+                meterName.StartsWith("OpenTelemetry", StringComparison.OrdinalIgnoreCase) ||
+                meterName.StartsWith("System.Diagnostics" + ".Metrics", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether instruments in the specified meter should be enabled based on the 
+        /// OpenTelemetry metrics configuration filters (include/exclude lists).
+        /// Follows the precedence logic:
+        /// 1. Built-in exclude list
+        /// 2. Customer configured include list  
+        /// 3. Customer configured exclude list
+        /// </summary>
+        /// <param name="meterName">The name of the meter to check</param>
+        /// <returns>True if the meter should be enabled; false otherwise</returns>
+        private bool ShouldEnableInstrumentsInMeterWithFilters(string meterName)
+        {
+            // First apply the built-in filtering logic
+            if (!ShouldEnableInstrumentsInMeter(meterName))
+            {
+                return false;
+            }
+
+            // Check if OpenTelemetry metrics + global OpenTelemetry are enabled
+            if (!_configuration.OpenTelemetryMetricsEnabled)
+            {
+                return false;
+            }
```

Comment:
Do we need this check here? I think that we stop the listener when that setting is disabled.